src/main/java/net/sf/jabref/logic/util/OS.java

```diff
@@ -11,4 +11,24 @@
     public static final boolean LINUX = osName.startsWith("linux");
     public static final boolean WINDOWS = osName.startsWith("win");
     public static final boolean OS_X = osName.startsWith("mac");
+    
+    public static final String guessProgramPath(String programName, String windowsDirectory) {
+        if (OS.WINDOWS) {
+            String progFiles = System.getenv("ProgramFiles(x86)");
```

Comment:
I'm quite sure the provided alternative is also OK (since it works and was in the original code, I stole the basic functionality from PushToTeXStudio).

In `openoffice.AutoDetectPaths` there is a method `findProgramFilesDir()` which use a manual method to find these two directories.

I believe that `guessProgramPath` could be improved quite a bit later on. Searching both locations and selecting the one where the file exists, search with arbitrary version numbers (at least WinEdt and OpenOffice have version numbers in their path) etc. Maybe it makes sense to get the full path in Linux as well? (Simply getting the result from `which xxx` would be a start...)